File: 프로그래머스/2/131704. 택배상자/택배상자.java
import java.util.*;

class Solution {
    public int solution(int[] order) {
        int i = 0;
        int conveyor = 0;
        Stack<Integer> stack = new Stack<>();
        
        while(i < order.length) {
            if(order[i] == conveyor) {
                conveyor++;
                i++;
            } else if(!stack.isEmpty() && order[i] == stack.peek()) {
                stack.pop();
                i++;
            } else {
                if(order[i] > conveyor) {
                    stack.push(conveyor);
                    conveyor++;
                } else {
                    break;
                }
            }
        }
        
        return i;
    }
}

File: 프로그래머스/2/131704. 택배상자/택배상자.py
from collections import deque

def solution(order):
    conveyor = 1
    stack = []
    i = 0
    while i < len(order):
        if conveyor == order[i]:
            i += 1
            conveyor += 1
        elif stack and stack[-1] == order[i]:
            i += 1
            stack.pop()
        else:
            if order[i] > conveyor:
                stack.append(conveyor)
                conveyor += 1
            else:
                break
    
    return i